### ISSUE TITLE
Deterministic actions using PPO

### ISSUE THREAD

**Issue by donamin** [openai/spinningup#190]
Hi

How can I get a deterministic action out of PPO policy? I need to turn off the exploration noise for that; but there doesn't seem to be such a switch in the code.

**Comment by Alder85** [comment]
By deterministic you mean one that directly uses the outputs of the network and doesn't use random normal/multinomial distributions? My best guess:
For discrete, in `mlp_categorical_policy` (in `ppo/core.py`), change `pi` to be the max output of the `logits` rather than roll a multinomial from them. This will always take the action with the highest confidence.
For continuous, in `mlp_gaussian_policy` (same file), change `pi` to be `mu` instead of generating a distribution centered around it. This will always take an action with the values outputted by the network.
This change will probably break the pi_loss update, but you can probably use it on a already-trained network without much issue.

**Comment by jachiam** [comment]
Hello! Indeed, the code isn't set up to support this use case currently. If you wanted to do it, there are a couple of ways I could think to make it happen:

- Before training, modify the PPO core file so that `mu` is also output by the `actor_critic` call. Then modify the PPO algorithm file so that `mu` gets saved by the logger. Then you can just train and enjoy.

- If you want to get a deterministic action out of an already-trained PPO policy where you **didn't** do that, you can try one of two things. Either dig through the computation graph for the output from `mu`, or find `log_std` and set all of its elements to `-inf`. 

To dig through for `mu`, I think the following should work:

```
sess = tf.InteractiveSession()
graph_dict = restore_tf_graph(sess, 'path/to/model')
pi = graph_dict['pi']
mu = pi.graph.get_tensor_by_name('pi/dense_2/BiasAdd:0')
```

That's assuming the MLP has two layers. Change `dense_2` to `dense_N` where `N` is the number of layers from your network.
